mlflow/utils/_spark_utils.py

```diff
@@ -36,6 +38,30 @@ def _prepare_subprocess_environ_for_creating_local_spark_session():
     os.environ.pop("PYSPARK_GATEWAY_SECRET", None)
 
 
+def _get_spark_scala_version_from_spark_session(spark):
+    version = Version(spark._jvm.scala.util.Properties.versionNumberString())
+    return f"{version.major}.{version.minor}"
+
+
+def _get_spark_scala_version():
+    from pyspark.sql import SparkSession
+
+    from mlflow.utils.databricks_utils import is_in_databricks_runtime
+
+    if is_in_databricks_runtime():
+        return os.environ["SPARK_SCALA_VERSION"]
+
+    if spark := _get_active_spark_session():
+        return _get_spark_scala_version_from_spark_session(spark)
+
+    try:
+        spark = SparkSession.builder.master("local[1]").getOrCreate()
```

Comment:
can we use a context manager?

Reply:
/ai Does using context manager like this ensure that the spark session is properly terminated on completion/exception?

```
with SparkSession.builder.master("local[1]").getOrCreate() as spark:
    // some operation

Reply:
@harupy I was stuck on this a while (since last month we talked about spark session termination haha), but I think the current way of using context manager for SparkSession is not sufficient. I think we need to add __exit__ handler to make it properly stop.